pandas/core/arrays/datetimes.py

```diff
@@ -404,8 +393,15 @@ def _generate_range(
                     start = start.tz_localize(None)
                 if end is not None:
                     end = end.tz_localize(None)
+
             # TODO: consider re-implementing _cached_range; GH#17914
-            values, _tz = generate_regular_range(start, end, periods, freq)
+            if isinstance(freq, Tick):
+                values = generate_time_range(start, end, periods, freq)
+            else:
+                xdr = generate_range(start=start, end=end, periods=periods, offset=freq)
+                values = np.array([x.value for x in xdr], dtype=np.int64)
+
+            _tz = start.tz if start is not None else end.tz
```

Comment:
@jbrockmendel: This piece of code used to be in the "old" `generate_regular_range`. I can bring back a wrapper function that I created in one of the previous commit if you think that is necessary.